cpp/src/arrow/compute/compute-test.cc

```diff
@@ -236,6 +236,65 @@ TEST_F(TestCast, ToIntDowncastSafe) {
   // underflow
   vector<int32_t> v6 = {0, 1000, 2000, -70000, 0};
   CheckFails<Int32Type>(int32(), v6, is_valid, int16(), options);
+
+  vector<int32_t> v7 = {0, 1000, 2000, -70000, 0};
+  CheckFails<Int32Type>(int32(), v7, is_valid, uint8(), options);
+}
+
+TEST_F(TestCast, IntegerSignedToUnsigned) {
+  CastOptions options;
+  options.allow_int_overflow = false;
+
+  vector<bool> is_valid = {true, false, true, true, true};
+
+  vector<int32_t> v1 = {INT32_MIN, 100, -1, 0, INT32_MAX};
+  // Same width
+  CheckFails<Int32Type>(int32(), v1, is_valid, uint32(), options);
+  // Wider
+  CheckFails<Int32Type>(int32(), v1, is_valid, uint64(), options);
+  // Narrower
+  CheckFails<Int32Type>(int32(), v1, is_valid, uint16(), options);
+
+  options.allow_int_overflow = true;
+
+  vector<uint32_t> v2 = {(uint32_t)INT32_MIN, 100, (uint32_t)-1, 0, (uint32_t)INT32_MAX};
```

Comment:
Use `static_cast` instead of C-style casts. Ordinarily cpplint catches this so I'm not sure why it did not here